dulwich/porcelain.py

```diff
@@ -2222,7 +2223,10 @@ def describe(repo, abbrev=7):
 
         # If there are no tags, return the current commit
         if len(sorted_tags) == 0:
-            return f"g{find_unique_abbrev(r.object_store, r[r.head()].id)}"
+            object_id = r[r.head()].id
+            if abbrev is not None:
+                return object_id[abbrev_slice].decode("ascii")
```

Comment:
In fact: I think there's a bug here; this should be decoding _before_ the substring range-access.  Fixing that up momentarily.

Reply:
And it's also missing the `g` prefix.  A little more work to do here after all.